syntax = "proto3";

service ArtistGrpcService {
  rpc GetAll (Empty) returns (stream ArtistProto) {}
  rpc GetById (ArtistIdProto) returns (ArtistProto) {}
  rpc Insert (InsertArtistProto) returns (Empty) {}
  rpc Delete (ArtistIdProto) returns (Empty) {}
}

service RecordGrpcService {
    rpc GetAll (Empty) returns (stream RecordProto) {}
    rpc GetById (RecordIdProto) returns (RecordProto) {}
    rpc Insert (InsertRecordProto) returns (Empty) {}
    rpc UpdateQuantity (UpdateQuantityRequestProto) returns (Empty) {}
    rpc UpdatePrice (UpdatePriceRequestProto) returns (Empty) {}
    rpc Delete (RecordIdProto) returns (Empty) {}
}

service LabelGrpcService {
  rpc GetAll (Empty) returns (stream LabelProto) {}
  rpc GetById (LabelIdProto) returns (LabelProto) {}
  rpc Insert (InsertLabelProto) returns (Empty) {}
  rpc Delete (LabelIdProto) returns (Empty) {}
}

message Empty {}

message ArtistProto {
  string id = 1;
  string artistName = 2;
  string country = 3;
  string realName = 4;
}

message InsertArtistProto {
  string artistName = 1;
  string country = 2;
  string realName = 3;
}

message ArtistIdProto {
  string id = 1;
}

message LabelProto {
  string id = 1;
  string name = 2;
  string country = 3;
}

message InsertLabelProto {
  string name = 1;
  string country = 2;
}

message LabelIdProto {
  string id = 1;
}


message RecordProto {
    string id = 1;
    string title = 2;
    ArtistProto artist = 3;
    string genre = 4;
    string style = 5;
    string releaseDate = 6;
   LabelProto label = 7;
    string country = 8;
    double price = 9;
    int32 quantity = 10;
}

message InsertRecordProto {
    string title = 1;
    string artistId = 2;
    string genre = 3;
    string style = 4;
    string releaseDate = 5;
    string labelId = 6;
    string country = 7;
    double price = 8;
    int32 quantity = 9;
}

message RecordIdProto {
    string id = 1;
}

message UpdateQuantityRequestProto {
    string id = 1;
    int32 quantity = 2;
}

message UpdatePriceRequestProto {
    string id = 1;
    double price = 2;
}
